service-base/src/test/java/org/eclipse/hono/service/VertxBasedHealthCheckServerTest.java

```diff
@@ -0,0 +1,180 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.hono.service;
+
+import io.vertx.core.Future;
+import io.vertx.core.Vertx;
+import io.vertx.ext.healthchecks.HealthCheckHandler;
+import io.vertx.ext.web.client.WebClient;
+import io.vertx.ext.web.client.WebClientOptions;
+import io.vertx.ext.web.client.predicate.ResponsePredicate;
+import org.eclipse.hono.config.ServerConfig;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.Timeout;
+import org.junit.runner.RunWith;
+
+import io.vertx.ext.unit.Async;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+
+import java.net.HttpURLConnection;
+
+/**
+ * Test cases verifying the behavior of {@code VertxBasedHealthCheckServer}.
+ */
+@RunWith(VertxUnitRunner.class)
+public class VertxBasedHealthCheckServerTest {
+
+    private static final String HOST = "localhost";
+    private static Vertx vertx;
+
+    /**
+     * Time out each test after five seconds.
+     */
+    @Rule
+    public final Timeout timeout = Timeout.seconds(5);
+
+    /**
+     * Sets up Vert.x for the test.
+     */
+    @BeforeClass
+    public static void init() {
+        vertx = Vertx.vertx();
+    }
+
+    /**
+     * Tests that a health check server can be configured with an insecure port.
+     *
+     * @param ctx The test context.
+     */
+    @Test
+    public void testHealthCheckServerWithInsecureEndpoint(final TestContext ctx) {
+        final ServerConfig config = new ServerConfig();
+        config.setInsecurePortEnabled(true);
+        config.setInsecurePort(0);
+        config.setInsecurePortBindAddress(HOST);
+
+        final VertxBasedHealthCheckServer server = new VertxBasedHealthCheckServer(vertx, config);
+        registerHealthChecks(ctx, server, 1);
+
+        server.start().compose(result -> Future.succeededFuture(getWebClient(server, false)))
+                .compose(httpClient -> checkHealth(ctx, httpClient, "/liveness"))
+                .compose(httpClient -> checkHealth(ctx, httpClient, "/readiness"))
+                .compose(result -> server.stop())
```

Comment:
IMHO the server should be stopped in any case, i.e. the server should be stopped in a separate method that is annotated with `@AfterEach`

Reply:
So the suggestion is to introduce a member variable which holds the current `VertxBasedHealthCheckServer` and in the `@AfterEach` annotated method this server is stopped if it exists?

Reply:
That should do the trick, shouldn't it?